terraform/deployments/cluster-infrastructure/variables.tf

```diff
@@ -3,11 +3,32 @@ variable "govuk_aws_state_bucket" {
   description = "The name of the S3 bucket used for govuk-aws's Terraform state files."
 }
 
+variable "cluster_name" {
+  type        = string
+  description = "Name for the EKS cluster."
+  default     = "govuk"
+}
+
 variable "cluster_log_retention_in_days" {
   type        = number
   description = "Number of days to retain cluster log events in CloudWatch."
 }
 
+variable "eks_control_plane_subnets" {
+  type        = map(object({ az = string, cidr = string }))
+  description = "Map of {subnet_name: {az=<az>, cidr=<cidr>}} for the public subnets for the EKS cluster's apiserver."
```

Comment:
This is probably something to address later, but I think we probably want to consider keeping the API server endpoint private, given the potential consequences of a leaked EKS credential or security issue with k8s/EKS, or whether relying on CIDR/IP whitelisting is good enough - that obviously raises the question of what a non-public network route to the API endpoint would be (VPN, bastion, etc.). Similarly we probably should make use of the private API endpoint for in-cluster API calls, if we're not already.

([public/private API endpoint docs](https://docs.aws.amazon.com/eks/latest/userguide/cluster-endpoint.html))

Reply:
Yeah those are both good points. I've made a note to look into both (private apiserver and whether we need to do anything about the routing of traffic between nodes and apiserver).

My hunch is that with out current AWS setup, making the apiserver private would just move the problem rather than provide any actual security improvement - e.g. we'd end up sshing into a bastion VM or something ugly like that. We could stick an IP allowlist on the apiserver (and we'll almost certainly need to do so for prod - that'd be roughly equivalent to what we have now).

Reply:
Private apiserver endpoint wasn't enabled before but it is now - thanks for picking up on that 👍